bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Future.java

```diff
@@ -26,13 +31,19 @@
  */
 public class Future {
     private final Strand strand;
+    private final AtomicBoolean visited = new AtomicBoolean(false);
 
      Future(Strand strand) {
         this.strand = strand;
     }
 
     public void complete(Object returnValue) {
+        if (visited.get()) {
```

Comment:
Can we return, if the strand is already unblocked?

Reply:
+1 to the current design as its good to have an error so API user knows there are multiple complete calls.